add helper for computing noise_multiplier from epsilon

jagielski opened the pull request:
A common workflow starts from a privacy budget, and requires obtaining the noise multiplier that is required for DP-SGD to satisfy the given privacy budget. With the current codebase, this requires running a binary search. This pull request implements this binary search. It can be found in the analysis/compute_noise_from_budget.py script, using the same API as the existing analysis/compute_dp_sgd_privacy.py.

schien1729 commented on tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py:
```diff
@@ -0,0 +1,67 @@
+Sampled Gaussian Mechanism. The mechanism's parameters are controlled by flags.
+
+Example:
+  compute_dp_sgd_privacy
```
Please update this command line to reflect name of program.

schien1729 commented on tensorflow_privacy/privacy/analysis/compute_noise_from_budget_lib.py:
```diff
@@ -0,0 +1,80 @@
+  """Compute noise based on the given hyperparameters."""
+  q = batch_size / n  # q - the sampling ratio.
+  if q > 1:
+    raise app.UsageError('n must be larger than the batch size.')
```
Minor note: Given that n is usually fixed, while the batch size can be varied, maybe rephrase this to say that batch size must be at most n?

schien1729 commented on tensorflow_privacy/privacy/analysis/compute_noise_from_budget.py:
```diff
@@ -0,0 +1,67 @@
+flags.DEFINE_float('delta', 1e-6, 'Target delta')
+
+
+def main(argv):
```
As I read it, in the rare case that noise of 1e-5 is too much, there's not much the user can do to use this tool. Can you either document that, or maybe add a minimum noise argument to compute_noise, defaulting to 1e-5?

schien1729 reviewed (changes requested):
Thanks, Matthew! This is indeed a useful tool that users have requested. I've made some minor suggestions-- sorry it took so long.

jagielski commented:
Thanks for the comments, Steve! I've addressed each of your comments in the latest commit. On the last one, I've elected to add a min_noise parameter, and I've added a test to check that if min_noise is too large, it has the expected behavior (just return 0).
